sentinel-ai-session/src/test/java/com/phonepe/sentinel/session/AgentSessionExtensionTest.java

```diff
@@ -155,6 +174,76 @@ void test(final WireMockRuntimeInfo wiremock) {
+        final var toolbox = new TestToolBox("Santanu");
+        final var model = new SimpleOpenAIModel(
+                "gpt-4o",
+                SimpleOpenAIAzure.builder()
+//                        .baseUrl(EnvLoader.readEnv("AZURE_ENDPOINT"))
+//                        .apiKey(EnvLoader.readEnv("AZURE_API_KEY"))
+                        .baseUrl(wiremock.getHttpBaseUrl())
+                        .apiKey("BLAH")
+                        .apiVersion("2024-10-21")
+                        .objectMapper(objectMapper)
+                        .clientAdapter(new OkHttpClientAdapter(new OkHttpClient.Builder().build()))
+                        .build(),
+                objectMapper
+        );
+
+
+        InMemoryHistoryStore historyStore = new InMemoryHistoryStore();
+        final var agent = SimpleAgent.builder()
+                .setup(AgentSetup.builder()
+                        .mapper(objectMapper)
+                        .model(model)
+                        .modelSettings(ModelSettings.builder()
+                                .temperature(0.1f)
+                                .seed(1)
+                                .build())
+                        .build())
+                .extensions(List.of(AgentSessionExtension.<UserInput, String, SimpleAgent>builder()
+                        .historyStore(historyStore)
+                        .setup(AgentSessionExtensionSetup.builder().mode(AgentSessionExtensionMode.HISTORY).build())
+                        .mapper(objectMapper)
+                        .build()))
+                .build()
+                .registerToolbox(toolbox);
+
+        final var requestMetadata = AgentRequestMetadata.builder()
+                .sessionId("s1")
+                .userId("ss")
+                .build();
+        final var response = agent.execute(
+                AgentInput.<UserInput>builder()
+                        .request(new UserInput("Hi"))
+                        .requestMetadata(requestMetadata)
+                        .build());
+        log.info("Agent response: {}", response.getData());
+
+        //Thread.sleep(1000);
+        //assertEquals(1, historyStore.history("s1").orElseThrow().getMessages().size());
+
+        final var response2 = agent.execute(
+                AgentInput.<UserInput>builder()
+                        .request(new UserInput("How is the weather at user's location?"))
+                        .requestMetadata(requestMetadata)
+                        //.oldMessages(response.getAllMessages())
+                        .build());
+        log.info("Second call: {}", response2.getData());
+        if (log.isTraceEnabled()) {
+            log.trace("Messages: {}", objectMapper.writerWithDefaultPrettyPrinter()
+                    .writeValueAsString(response2.getAllMessages()));
+        }
+
+        //Thread.sleep(1000);
+        //assertEquals(2, historyStore.history("s1").orElseThrow().getMessages().size());
```

Comment:
Fix tests and ensure messages are being save. Add other tests to ensure features are being turned on and off properly, messages are getting filtered when saving and so on